scripts/run_sample_pipeline.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# This script executes the sample pipeline in the example folder, checks the correct execution and
+# cleans up the directory again
+set -e
+
+cleanup() {
+  # Create a temporary directory
+  artifact_directory="./examples/sample_pipeline_test/.artifacts"
+  # Check if the temporary directory exists before attempting to remove it
+  if [ -d "$artifact_directory" ]; then
+    rm -r "$artifact_directory"
+    echo "Temporary directory removed"
+  fi
+}
```

Comment:
I still think this is needed.

You might even want to catch the original exit code and reraise it like this:

```bash
cleanup() {
    rv=$?
    ...
    exit $rv
}
```

Reply:
I wasn't able to get the exit code of the failed docker container execution. I only get the exit code of the docker compose process. In particular, if a component fails with 1, and the compose succeeded with 0, I will miss the 1 since the docker compose process was completed successfully.

Reply:
I've added it. Now we catch exit codes that are not coming from the component code. Might be fine here, cause we are testing not the component code itself.

Reply:
What exit codes are we catching then? Because we do want to catch errors from eg. the executor, which is also running in the container.

Reply:
If docker compose is always exiting with code 0, we might want to look into the `--abort-on-container-exit` or `--exit-code-from` [options](https://docs.docker.com/engine/reference/commandline/compose_up/#options).